app/src/main/java/com/korneysoft/rsschool2021_android_task_6_musicapp/ui/MainActivity.kt

```diff
@@ -0,0 +1,263 @@
+package com.korneysoft.rsschool2021_android_task_6_musicapp.ui
+
+import android.graphics.drawable.Drawable
+import android.os.Bundle
+import android.support.v4.media.session.PlaybackStateCompat
+import android.text.method.ScrollingMovementMethod
+import android.view.View
+import android.widget.SeekBar
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.request.RequestListener
+import com.bumptech.glide.request.target.Target
+import com.korneysoft.rsschool2021_android_task_6_musicapp.R
+import com.korneysoft.rsschool2021_android_task_6_musicapp.application.MyApplication
+import com.korneysoft.rsschool2021_android_task_6_musicapp.data.Track
+import com.korneysoft.rsschool2021_android_task_6_musicapp.databinding.ActivityMainBinding
+import com.korneysoft.rsschool2021_android_task_6_musicapp.utils.msecToTime
+import com.korneysoft.rsschool2021_android_task_6_musicapp.viewmodel.MainViewModel
+import com.korneysoft.rsschool2021_android_task_6_musicapp.viewmodel.PlayerControl
+import javax.inject.Inject
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+
+    private var isSeekBarTrackingTouch = false
+    private var currentTrack: Track? = null
+
+    @Inject
+    lateinit var model: MainViewModel
+
+    private var playerControl: PlayerControl? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        (application as MyApplication).appComponent.inject(this)
+
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // enable auto-scrolling for textLog
+        binding.textLog.movementMethod = ScrollingMovementMethod()
+
+        if (model is PlayerControl) {
+            playerControl = model
+        }
+
+        setListenersPlayerControl()
+        setListenerSeekBar()
+        registerObservers()
+        prefatorySetSeekBarSettings()
+        showCurrentTrackInfo()
+    }
+
+    private fun registerObservers() {
+        model.playerEventLiveData.observe(this,
```

Comment:
model.playerEventLiveData.observe(this, {
            it?.let {
                playerStateApply(it)
            }
        })

Reply:
I used  construction " it ?: return " to avoid unnecessary nesting.
But if it not best practice, ok I fixed it.